src/state/nftMarket/reducer.ts

```diff
@@ -29,6 +36,23 @@ export const fetchNftsFromCollections = createAsyncThunk<NFT[], string>(
   },
 )
 
+export const fetchUserNfts = createAsyncThunk<NftSubgraphEntity[], { account: string; profileNftTokenId: string }>(
+  'nft/fetchUserNfts',
+  async ({ account, profileNftTokenId }) => {
+    const nftIdsInWallet = await fetchWalletNftIds(account)
+    if (profileNftTokenId) {
+      nftIdsInWallet.push(profileNftTokenId)
+    }
+    const marketDataForNftsInWallet = await getNftsMarketData({ tokenId_in: nftIdsInWallet })
+    const walletNftsWithMarketData = nftIdsInWallet.map((walletId) => {
+      const marketData = marketDataForNftsInWallet.find((marketNft) => marketNft.tokenId === walletId)
+      return marketData || { tokenId: walletId, metadataUrl: null, transactionHistory: null }
+    })
+    const marketDataForNftsOnSale = await getNftsMarketData({ currentSeller: account.toLowerCase() })
+    return [...walletNftsWithMarketData, ...marketDataForNftsOnSale]
```

Comment:
This returns _very_ simple data for now for, until we have the API and know more about how collection metadata is retrieved. 

It's just an array of objects with a `tokenId`, a `metaDataUrl` and a `transactionHistory` object